### PR TITLE
Add a do-while loop around macros.

### PR DESCRIPTION
**Pull request by DemiMarie**
## Description

Add do-while loops around various macros.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required - no functional changes
- [x] **backport** in #7721
- [x] **tests** not required - no functional changes

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

**Review by bensze01 — APPROVED** [review]
Looks good to me. Please rebase on top of development to fix the CI issues (See #7699), and backport to 2.28.

**Comment by gilles-peskine-arm** [comment]
Thank you for your contributions. I don't have time to review everything, and I'm afraid I can't commit to review all of your pull requests. In general, please don't request review from specific people except by prior agreement.

**Comment by DemiMarie** [comment]
> Thank you for your contributions.

You’re welcome!

> I don't have time to review everything, and I'm afraid I can't commit to review all of your pull requests.

That’s not a problem.

> In general, please don't request review from specific people except by prior agreement.

Sorry about that.  Is there a way to disable the GitHub feature that suggests people to request review from?

File: library/x509.c [hunk 1]
```diff
@@ -1700,16 +1704,18 @@ int mbedtls_x509_info_subject_alt_name(char **buf, size_t *size,
...
+        sep = ", ";                                     \
+    } while (0)
 
-#define CERT_TYPE(type, name)                    \
-    if (ns_cert_type & (type))                 \
-    PRINT_ITEM(name);
+#define CERT_TYPE(type, name)                           \
+    do {                                                \
+        if (ns_cert_type & (type))                      \
+            PRINT_ITEM(name);                           \
```
Comment (bensze01) [inline library/x509.c:1717]: The new code style mandates brackets around one-line `if`` statements. This should allow the `check_code_style`  unit-test to pass.
```suggestion
        if (ns_cert_type & (type)) {                    \
            PRINT_ITEM(name);                           \
        }                                               \
```

File: library/x509.c [hunk 2]
```diff
@@ -1734,9 +1740,11 @@ int mbedtls_x509_info_cert_type(char **buf, size_t *size,
     return 0;
 }
 
-#define KEY_USAGE(code, name)    \
-    if (key_usage & (code))    \
-    PRINT_ITEM(name);
+#define KEY_USAGE(code, name)      \
+    do {                           \
+        if ((key_usage) & (code))  \
+            PRINT_ITEM(name);      \
```
Comment (bensze01) [inline library/x509.c:1746]: Ditto.
```suggestion
        if ((key_usage) & (code))  { \
            PRINT_ITEM(name);        \
        }                            \
```

**Review by bensze01 — CHANGES_REQUESTED** [review]

**Review by bensze01 — APPROVED** [review]
LGTM

**Comment by bensze01** [comment]
The readthedocs build completed successfully, but failed to update its status on github, for some reason.

**Pull request merged**